### PR TITLE
Adding the Acosh, Asinh, Atanh, and Cbrt functions to System.Math and System.MathF

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves https://github.com/dotnet/corefx/issues/16428

### CODE REVIEW

**Comment by tannergooding** [comment]
FYI. @eerhardt, @jkotas

**Comment by tannergooding** [comment]
Is there an explicit step I need to take to get UWP updated?

**Comment by jkotas** [comment]
>Is there an explicit step I need to take to get UWP updated?

It needs to be implemented in ProjectN. Math is not in shared CoreLib partition.

**Comment by tannergooding** [comment]
I have a CodeFlow review up which adds the ProjectN changes. I currently have @jkotas, @janvorli, and ianb tagged on it.

Let me know if I should tag anyone else.

**Comment by karelz** [comment]
@tannergooding what's left to finish the PR?

**Comment by tannergooding** [comment]
@karelz, as far as I know, the PR itself is done.

There was some additional work required to support CoreRT which is pending checkin (I've added you to the internal e-mail thread, since this requires changes to the TFS side of things).

**Comment by karelz** [comment]
@tannergooding where do we stand on this PR? If something is blocking it, let's close it and reopen when it is ready ...

**Comment by tannergooding** [comment]
Currently pending a response from @cshung on the internal thread. The CoreRT changes are ready to merge, but is blocked due to additional changes required to the debugger side.

Closing for now. Will reopen after all CoreRT changes are finalized.

**Comment by cshung** [comment]
I am sorry if I am not making it clear.

There are pending work on the debugger side to handle these new intrinsic functions, but there is no need to block on the debugger side work to merge any changes in.

Once the change is merged in I can get started on getting the debugger to work with them.

**Comment by tannergooding** [comment]
CoreRT changes are finally in. This should be good now.

**Comment by tannergooding** [comment]
FYI. @eerhardt 

This is unblocked now. Could you review when you get the chance?

File: src/System.Runtime.Extensions/tests/System/MathF.netcoreapp.cs [hunk 1]
```diff
@@ -431,6 +496,86 @@ public static void Atan2_IEEE_Legacy(float y, float x, float expectedResult, flo
...
+        [InlineData( 0.761594156f,            1.0f,                   CrossPlatformMachineEpsilon * 10)]
+        [InlineData( 0.810463806f,            1.12837917f,            CrossPlatformMachineEpsilon * 10)]  // expected:  (2 / sqrt(pi))
+        [InlineData( 0.888385562f,            1.41421356f,            CrossPlatformMachineEpsilon * 10)]  // expected:  (sqrt(2))
+        [InlineData( 0.894238946f,            1.44269504f,            CrossPlatformMachineEpsilon * 10)]  // expected:  (log2(e))
+        [InlineData( 0.917152336f,            1.57079633f,            CrossPlatformMachineEpsilon * 10)]  // expected:  (pi / 2)
+        [InlineData( 0.980198020f,            2.30258509f,            CrossPlatformMachineEpsilon * 10)]  // expected:  (ln(10))
+        [InlineData( 0.991328916f,            2.71828183f,            CrossPlatformMachineEpsilon * 10)]  // expected:  (e)
+        [InlineData( 0.996272076f,            3.14159265f,            CrossPlatformMachineEpsilon * 10)]  // expected:  (pi)
+        [InlineData( 1.0f,                    float.PositiveInfinity, 0.0f)]
```
Comment (eerhardt) [inline src/System.Runtime.Extensions/tests/System/MathF.netcoreapp.cs]: How about a test for 2, or some other value where 1 < value < Infinity?

File: src/System.Runtime.Extensions/tests/System/MathF.netcoreapp.cs [hunk 2]
```diff
@@ -240,6 +240,32 @@ public static void Acos(float value, float expectedResult, float allowedVariance
             AssertEqual(expectedResult, MathF.Acos(value), allowedVariance);
         }
 
+        [Theory]
+        [InlineData( float.NegativeInfinity,  float.NaN,              0.0f)]
+        [InlineData(-0.0f,                    float.NaN,              0.0f)]
+        [InlineData( float.NaN,               float.NaN,              0.0f)]
+        [InlineData( 0.0f,                    float.NaN,              0.0f)]
+        [InlineData( 1.0f,                    0.0f,                   CrossPlatformMachineEpsilon)]
```
Comment (eerhardt) [inline src/System.Runtime.Extensions/tests/System/MathF.netcoreapp.cs]: Similar test here for a value between 0 and 1.  and a negative value that isn't negative zero.

**Comment by tannergooding** [comment]
@eerhardt, do you mind if I update the tests in a follow up PR.

Most of the Math tests are missing values in the requested scenarios right now and I'd like to update them all at once.

**Comment by eerhardt** [comment]
Yes, that would be fine with me.  Thanks for following up to look at the other math tests.

**Comment by tannergooding** [comment]
Logged https://github.com/dotnet/corefx/issues/26941

**Comment by briansull** [comment]
These new tests hit an unreach assert in the JIT:

https://github.com/dotnet/coreclr/issues/16272

**Pull request merged**